[DOC] Restructure Pyroscope documentation and share content (#3798)
grafana/pyroscope: docs/sources/shared/intro/continuous-profiling.md

---
headless: true
description: Shared file for intro to continuous profiling.
---

[//]: # 'When to use continuous profiling.'
[//]: # 'This shared file is included in these locations:'
[//]: # '/pyroscope/docs/sources/introduction/continuous-profiling.md'
[//]: # '/website/content/grafana-cloud/monitor-applications/profiles/introduction/continuous-profiling.md'
[//]: #
[//]: # 'If you make changes to this file, verify that the meaning and content are not changed in any place where the file is included.'
[//]: # 'Any links should be fully qualified and not relative: /docs/grafana/ instead of ../grafana/.'

<!--  When to use continuous profiling -->

Continuous profiling is a systematic method of collecting and analyzing performance data from production systems.

> Traditionally, profiling is used to debug application on an as-needed basis.

Reviewer: ```suggestion
Traditionally, profiling is used to debug applications on an as-needed basis.
```

You are probably used to running a benchmark tool locally and getting a `pprof` file in Go or maybe connecting into a misbehaving prod instance and pulling a flame graph from a `JFR` file in Java.

> This is great for debugging but not so great for production.

Reviewer 2: I like the use of conversational tone, but this might border on colloquial.

> Traditionally, profiling is used as a free-form debugging tool in languages like Go and Java.
> You are probably used to running a benchmark tool locally and getting a `pprof` file in Go or maybe connecting into a misbehaving prod instance and pulling a flame graph from a `JFR` file in Java.
> This is great for debugging but not so great for production.

Author: How's this? 
```suggestion
Traditionally, profiling is a free-form debugging tool in languages like Go and Java.
For example, you can run a benchmark tool locally and get a `pprof` file in Go or connect to a misbehaving prod instance and pull a flame graph from a `JFR` file in Java.
This method is good for debugging, but not robust enough for production.
```

Reviewer: For the first sentence, perhaps something like "Traditionally, we use profiling to debug our applications on-demand." or something even more precise like "Traditionally, we use profiling to debug our applications on an as-needed basis." could also work?

The other changes look good to me.

Author: Thank you for the suggestion. I've reworded slightly because we try not to use "we" or "I" (first person) in our writing except for blog posts and release notes.

2. **Complements other observability tools:** Continuous profiling fills critical gaps left by metrics, logs, and tracing, creating a more comprehensive observability strategy.
3. **Proactive performance optimization:** Regular profiling enables teams to proactively identify and resolve performance bottlenecks, leading to more efficient and reliable applications.
![Diagram showing 3 benefits of continuous profiling](https://grafana.com/static/img/pyroscope/profiling-use-cases-diagram.png)

## Use cases

Adopting continuous profiling with tools like Grafana Pyroscope and Explore Profiles can lead to significant business advantages:

1. **Reduced operational costs:** Optimization of resource usage can significantly cut down cloud and infrastructure expenses
2. **Reduced latency:** Identifying and addressing performance bottlenecks leads to faster and more efficient applications
3. **Enhanced incident management:** Faster problem identification and resolution, reducing Mean Time to Resolution (MTTR) and improving end-user experience

![Infographic illustrating key business benefits](https://grafana.com/static/img/pyroscope/cost-cutting-diagram.png)

> ### Reduce operational costs

Reviewer 2: ```suggestion
### Reduced operational costs
```
